test/local/routes/recovery-codes.js

```diff
@@ -20,15 +20,10 @@ function runTest(routePath, requestOptions) {
   route = getRoute(routes, routePath)
   request = mocks.mockRequest(requestOptions)
   request.emitMetricsEvent = sinon.spy(() => P.resolve({}))
-  return new P((resolve, reject) => {
-    route.handler(request, response => {
-      if (response instanceof Error) {
-        reject(response)
-      } else {
-        resolve(response)
-      }
-    })
-  })
+
+  return route.handler(request)
+    .then((res) => { return res })
```

Comment:
There are a few places where this is done, can you update them from comment above.